addon/registry.js

```diff
@@ -0,0 +1,57 @@
+let REGISTERED_RECEIVERS = {};
+
+export function registerReceiver(component) {
+  let componentName = getComponentRegistryKey(component);
+  REGISTERED_RECEIVERS[componentName] = component;
```

Comment:
Another alternative is to map the name to multiple instances and send the action to all matching ones that are registered.